Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,66 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
-                                Py_TYPE(self)->tp_name,
-                                GET_TD_DAYS(self));
+    PyObject *args = PyList_New(0);
+
+    if (args == NULL) {
+        return NULL;
+    }
+
+    if (GET_TD_DAYS(self) != 0) {
+        PyObject *days = PyUnicode_FromFormat("days=%d", GET_TD_DAYS(self));
+        if (days == NULL || PyList_Append(args, days) < 0) {
+            Py_DECREF(args);
+            Py_XDECREF(days);
+            return NULL;
+        }
+        Py_DECREF(days);
+    }
+
+
+    if (GET_TD_SECONDS(self) != 0 ||
+        (GET_TD_DAYS(self) == 0 && GET_TD_MICROSECONDS(self) == 0)) {
+        PyObject *seconds = PyUnicode_FromFormat("seconds=%d",
+                                                 GET_TD_SECONDS(self));
+        if (seconds == NULL || PyList_Append(args, seconds) < 0) {
+            Py_DECREF(args);
+            Py_XDECREF(seconds);
+            return NULL;
+        }
+        Py_DECREF(seconds);
+    }
+
+
+    if (GET_TD_MICROSECONDS(self) != 0) {
+        PyObject *microseconds = PyUnicode_FromFormat("microseconds=%d",
+                                                      GET_TD_MICROSECONDS(self));
+        if (microseconds == NULL || PyList_Append(args, microseconds) < 0) {
+            Py_DECREF(args);
+            Py_XDECREF(microseconds);
+            return NULL;
```

Comment:
Please move "Py_DECREF(args); return NULL;" to a new "error:" label at the end, and use "goto error;" here, to factorize the code (duplicated 3 times).